src/core/utils/workspace.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { WORKSPACE_PATH_PREFIX } from './constants';
+import { IBasePath } from '../public';
+
+export const getWorkspaceIdFromUrl = (url: string, basePath?: string): string => {
+  const regexp = new RegExp(`^${basePath || ''}\/w\/([^\/]*)`);
+  const urlObject = new URL(url);
+  const matchedResult = urlObject.pathname.match(regexp);
+  if (matchedResult) {
+    return matchedResult[1];
+  }
+
+  return '';
+};
+
+export const cleanWorkspaceId = (path: string) => {
+  return path.replace(/^\/w\/([^\/]*)/, '');
+};
+
+export const formatUrlWithWorkspaceId = (url: string, workspaceId: string, basePath: IBasePath) => {
```

Comment:
Should we includes a try-catch block to handle potential errors during URL parsing and formatting? Thanks

Reply:
I think so, the `new URL` may gives some unexpected error.